Source/ViewerController.swift

```diff
@@ -136,53 +136,53 @@ public class ViewerController: UIViewController {
 
         if presented {
             self.scrollView.configure()
-            if !self.collectionView.indexPathsForVisibleItems().contains(self.currentIndexPath) {
-                self.collectionView.scrollToItemAtIndexPath(self.currentIndexPath, atScrollPosition: .Bottom, animated: true)
+            if !self.collectionView.indexPathsForVisibleItems.contains(self.currentIndexPath) {
+                self.collectionView.scrollToItem(at: self.currentIndexPath, at: .bottom, animated: true)
             }
         }
     }
 
-    public override func viewDidAppear(animated: Bool) {
+    public override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
 
-        self.present(self.initialIndexPath, completion: nil)
+        self.present(with: self.initialIndexPath, completion: nil)
     }
 }
 
 extension ViewerController {
     #if os(iOS)
-    public override func prefersStatusBarHidden() -> Bool {
-        let orientation = UIApplication.sharedApplication().statusBarOrientation
+    public override var prefersStatusBarHidden : Bool {
+        let orientation = UIApplication.shared.statusBarOrientation
         if UIInterfaceOrientationIsLandscape(orientation) {
             return true
         }
 
         return self.shouldHideStatusBar
     }
 
-    public override func preferredStatusBarStyle() -> UIStatusBarStyle {
+    public override var preferredStatusBarStyle : UIStatusBarStyle {
         if self.shouldUseLightStatusBar {
-            return .LightContent
+            return .lightContent
         } else {
-            return self.presentingViewController?.preferredStatusBarStyle() ?? .Default
+            return self.presentingViewController?.preferredStatusBarStyle ?? .default
         }
     }
     #endif
 
     private func presentedViewCopy() -> UIImageView {
         let presentedView = UIImageView()
-        presentedView.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
-        presentedView.contentMode = .ScaleAspectFill
+        presentedView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        presentedView.contentMode = .scaleAspectFill
         presentedView.clipsToBounds = true
 
         return presentedView
     }
 
-    private func findOrCreateViewerItemController(indexPath: NSIndexPath) -> ViewerItemController {
+    fileprivate func findOrCreateViewerItemController(_ indexPath: IndexPath) -> ViewerItemController {
         let viewerItem = self.controllerDataSource!.viewerController(self, itemAtIndexPath: indexPath)
         var viewerItemController: ViewerItemController
 
-        if let cachedController = self.viewerItemControllerCache.objectForKey(viewerItem.id) as? ViewerItemController {
+        if let cachedController = self.viewerItemControllerCache.object(forKey: viewerItem.id as NSString) {
```

Comment:
Looking like it, I tried to change it so NSCache gets String but it wants NSString since it conforms to protocol AnyObject.